ArtOfIllusion/src/artofillusion/Scene.java

```diff
@@ -455,27 +454,36 @@ public void addObject(Object3D obj, CoordinateSystem coords, String name, UndoRe
   public void addObject(ObjectInfo info, UndoRecord undo)
   {
     addObject(info, objects.size(), undo);
-    updateSelectionInfo();
   }
 
+  /** 
+   * Add a new object to the scene.  No undo record is added
+   */
+  public void addObject(ObjectInfo info)
+  {
+    addObject(info, objects.size(), (UndoRecord)null);
+  }
+  
   /** Add a new object to the scene in the specified position.  If undo is not null,
       appropriate commands will be added to it to undo this operation. */
 
   public void addObject(ObjectInfo info, int index, UndoRecord undo)
   {
     info.setId(nextID++);
     if (info.getTracks() == null)
-      {
-        info.addTrack(new PositionTrack(info), 0);
-        info.addTrack(new RotationTrack(info), 1);
-      }
+    {
+      info.addTrack(new PositionTrack(info), 0);
```

Comment:
There may be some cases where an object should not have any default tracks, such as a dummy object that is holding data for a plugin.

Reply:
This addObject method is only way to add object to Scene as objects field is private access only. And this method is always checks and and adds 2 tracks to object. In my mind adding this tracks to every obejct is not necessary at all

Reply:
Sorry, I thought that you had added these lines. I see now that they have been in the codebase for a very long time.